src/artrace.py

```diff
@@ -934,412 +978,72 @@ def model_tilt_test(slf, det, msarc, guesstilts=None, censpec=None, plotQA=False
-                #     tsta = (-cof[1] - np.sqrt(cof[1]**2 - 4.0*cof[0]*cof[2]))/(2.0*cof[0])
-                #     tstb = (-cof[1] + np.sqrt(cof[1]**2 - 4.0*cof[0]*cof[2]))/(2.0*cof[0])
-                #     if tsta >= xfit.min() and tsta <= xfit.max() and tstb >= xfit.min() and tstb <= xfit.max():
-                #         if abs(tsta-pcen) < abs(tstb-pcen): params, fail = [0.0,tsta,0.0], False
-                #         else: params, fail = [0.0,tstb,0.0], False
-                #     elif tsta >= xfit.min() and tsta <= xfit.max(): params, fail = [0.0,tsta,0.0], False
-                #     elif tstb >= xfit.min() and tstb <= xfit.max(): params, fail = [0.0,tstb,0.0], False
-                #     else: params, fail = [0,0,0], True
-                #     #params, fail = arutils.poly_to_gauss(arutils.func_fit(xfit, yfit, 'polynomial', 2))
-                # except:
-                #     params, fail = [0,0,0], True
-                #params, fail = arutils.gauss_lsqfit(xfit, yfit, arcdet[j])
-                #params, fail = [0.0,np.sum(xfit*yfit)/np.sum(yfit),0.0], False
-                #params, fail = arutils.gauss_fit(xfit, yfit, arcdet[j])
-                cc = np.correlate(ccyfit, yfit, mode='same')
-                #try:
-                #    params, fail = arutils.poly_to_gauss(arutils.func_fit(xfit, np.log(cc/np.max(cc)), 'polynomial', 2)[::-1])
-                #except:
-                #    params, fail = [0,0,0], True
-                params, fail = arutils.gauss_lsqfit(xfit, cc, 0.0)
-                centv = ccval + pcen - arcdet[j] - params[1]
-                xtfit[sz-k] = ordcen[arcdet[j],0]-k
-                if guesstilts is not None:
-                    shfit[sz-k] = guesstilts[arcdet[j],ordcen[arcdet[j],0]-k]*float(msarc.shape[0]-1.0)
-                ytfit[sz-k] = centv - shfit[sz-k]
-                etfit[sz-k] = 0.02
-                apfit[sz-k] = params[0]
-                if fail: mtfit[sz-k] = 1.0
-                else:
-                    pcen = int(0.5+centv)
-                    mtfit[sz-k] = 0.0
-            if offchip: continue # Don't use lines that go off the chip (could lead to a bad trace)
-            wmask = np.where(mtfit==0.0)
-#            if True and guesstilts is not None:
-#                plt.clf()
-#                plt.plot(xtfit[wmask], ytfit[wmask], 'bx')
-#                plt.show()
-#               try:
-#                   tcoeff = np.polynomial.polynomial.polyfit(xtfit[wmask],ytfit[wmask],1,w=1.0/mt)
-#               except:
-#                   tcoeff = np.polynomial.polynomial.polyfit(xtfit[wmask],ytfit[wmask],1)
-            if guesstilts is not None: ytfit[wmask] -= np.median(ytfit[wmask])
+            # Perform a robust polynomial fit to the traces
             wmsk, mcoeff = arutils.robust_polyfit(xtfit[wmask], ytfit[wmask]/(msarc.shape[0]-1.0),
                                                   slf._argflag['trace']['orders']['tiltorder'],
```

Comment:
polyscan is used to accurately follow the trace of each arc line. These traces allow us to interpolate between the arc lines. The coefficients of polyfit are used in the PCA, and are needed to extrapolate to the edges of the chip (in the dispersion direction). See lines 1145-1146 below.